draft-lenders-dns-cbor.md

```diff
@@ -627,12 +555,100 @@ dns-response = [
+~~~
+
+For name compression, a new packing table setup tag TBD28259 ('n' and 'c' in ASCII) for CBOR-packed {{-cbor-packed}} is defined.
+It provides an implicit text string suffix sequence table for shared items _V_ which is prepended to the existing table for shared items that apply to the entire tag TBD28259 (by default empty table).
+The this implicit (i.e. not explicitly represented) table _V_ is constructed as follows:
+Any coherent sequence of text strings encountered within the rump of tag TBD28259, as well as any of its non-empty suffixes, are added to the table as an array in depth-first order.
+Text string sequences within any tables for shared items or argument items within the rump MUST not be added to _V_.
+If a sequence for which an array is already in _V_ is encountered, a shared item reference _i_ to that array in V replaces this sequence.
+This shared item reference _i_ means: take the sequence from the array at _V_\[_i_\] and put it into the surrounding array in place of _i_.
+The resulting rump should look like referencing the _i_-th string (depth first) in the sequence.
+
+The "application/dns+cbor" media type comes with an optional parameter "packed".
+If it is not provided, it is assumed to be 0.
+With packed=0, any CBOR object `obj` marked by the "application/dns+cbor" media type MUST explicitly be understood as `TBD28259(obj)`, unless it is already `obj` itself is already tagged explicitly with TBD28259 as a whole.
+
+### Example
+
+Take the following content (note that this is intentionally not legal "application/dns+cbor" to illustrate generality).
+
+~~~ edn
+[
+  "www", "example", "org",
+  ["svc", "www", "example", "org"],
+  "org", "example", "org", 42,
+  "svc", "www", "example", "org", 42
+]
+~~~
+{: #fig:name-compression-example-unpacked title="Unpacked example for implicit text string suffix sequence compression."}
+
+This would generate the following virtual table _V_.
+
+~~~ edn
+[
+    ["www", "example", "org"],
+    ["example", "org"],
+    ["org"],
+    ["svc", "www", "example", "org"],
+    ["org", "example", "org"]
+]
+~~~
+{: #fig:name-compression-example-table title="Implicit table of shared items for the example."}
+
+Note that the sequence "org", "example", "org" is added at index 4 with leading "org", instead of referencing index 2 + index 1, as it is its own distinct suffix sequence.
+
+The packed representation would thus be:
+
+~~~ edn
+TBD28259(
+  [
+    ["www", "example", "org"],
+    ["svc", simple(0)],
+    "org", simple(1), 42,
+    simple(3), 42
+  ]
+)
+~~~
+{: #fig:name-compression-example-packed title="The packed representation of the example."}
+
+Note, with "application/dns+cbor;packed=0" the surrounding TBD28259 can be elided (even though the content would not be parsable as application/dns+cbor).
+
+With, e.g., table setup tag 113, further packing can be achieved by nesting table packing them with TBD28259.
+
+~~~ edn
+TBD28259(
+  113(
```

Comment:
```suggestion
113(
  TBD28259(
```

otherwise prepending the table from 113 would shift the TBD28259 ones.

Reply:
Which clashes with the requirement that we want to elude TBD28259 for dns+cbor, hum...

Reply:
> otherwise prepending the table from 113 would shift the TBD28259 ones.

Huh? From how I read my definition (and the definition for 113), the tables of the outer tag are prepended to the tables of the inner tag. I want the TBD28259 table _V_ to be the “start” of the shared item table.

Reply:
At least for 113, it is definitely the other way around. I guess the rationale is that you enable shorter references within smaller contexts (inside the rump of an inner tag).

Reply:
Are you sure? Maybe @cabo can help to alleviate the confusion? Does the outer setup tag prepend its table(s) to the inner tag's table(s) or the inner tag to its outer tag's?

Reply:
The main argument is really composability.  A CBOR data item that uses cbor-packed should still work when it is embedded in some other table setup.

Reply:
Ok, so in this case, if I understand correctly, the outer tag should append to the packing table(s) of the inner tag.

E.g., taking the example

```
113(
  TBD28259(
    [
      ["org", 42],
      [
        ["www", "example", simple(5)],
        ["svc", simple(0)],
        simple(5), simple(1), simple(6),
        simple(3), simple(6)
      ]
    ]
  )
)
```

would still work with

```
  TBD28259(
    [
        ["www", "example", "org"],
        ["svc", simple(0)],
        "org", simple(1), 42,
        simple(3), 42
    ]
  )
```

while the other way around 

```
TBD28259(
  113(
    [
      ["org", 42],
      [
        ["www", "example", simple(5)],
        ["svc", simple(0)],
        simple(5), simple(1), simple(6),
        simple(3), simple(6)
      ]
    ]
  )
)
```

would completely need to rewrite the `simple(i)` references (and not just the new ones) for `113(...)` if `TBD28259(...)` is taken away to make sense.

```
  113(
    [
      ["org", 42],
      [
        ["www", "example", simple(0)],
        ["svc", "www", "example", simple(0)],
        simple(0), "example", simple(0), sample(1),
        "svc", "www", "example", "org", sample(1)
      ]
    ]
  )
```

This makes sense to me. Will fix (and try to find a better wording to make the table setup clearer).